tracer/src/Datadog.Tracer.Native/calltarget_tokens.cpp

```diff
@@ -501,14 +514,17 @@ HRESULT CallTargetTokens::ModifyLocalSig(ILRewriter* reWriter, TypeSignature* me
     *callTargetStateToken = callTargetStateTypeRef;
     *exceptionToken = exTypeRef;
     *callTargetReturnToken = callTargetReturn;
+
     if (returnSignatureType != nullptr)
     {
-        *returnValueIndex = newLocalsCount - 4 - additionalLocalsCount;
+        *returnValueIndex = newLocalsCount - 6 - additionalLocalsCount;
     }
     else
     {
         *returnValueIndex = static_cast<ULONG>(ULONG_MAX);
     }
+    *exceptionValueIndex = newLocalsCount - 5 - additionalLocalsCount;
+    *exceptionVaueEndIndex = newLocalsCount - 4 - additionalLocalsCount;
```

Comment:
The two locals that have been added should be placed in `TracerTokens` (`tracer_tokens.cpp`) using the `AddAdditionalLocals` and `GetAdditionalLocalsCount` methods, as demonstrated in `debugger_tokens.cpp`: https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Tracer.Native/debugger_tokens.cpp#L253:#L283

It's important to note that the `CallTargetTokens` class should hold general locals related to calltarget instrumentation, while the `TracerTokens` and `DebuggerTokens` child classes should hold any extra locals specific to their respective instrumentations. The two locals that were added to `CallTargetTokens` will also affect the debugger's instrumentation, which is not desired.
(I know it's a bit confusing).

Reply:
yes thanks very much, I missed this architecture part. fixed in b9b2c8c95e0dc5c966184933b404f5d1ed7d06ba